sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -344,3 +358,13 @@ object ResolveDefaultColumns {
     }
   }
 }
+
+/**
+ * Represents a table with an underlying raw schema to use for resolving DEFAULT column references.
+ * The 'ignoreColumnsForInserts' field represents a set of column names to exclude from
+ * consideration when performing INSERT INTO commands, if any.
+ */
+trait TableWithRawSchemaForColumnDefaults {
```

Comment:
Should this be a mixin trait in DSV2?

Reply:
We discussed offline, it's probably more straightforward to have the separate trait that we only have to think about when it's relevant for specific sources than in the core DSV2 API.